distribution/docs/src/main/resources/_contents/configuring-from-configuration-files-contents.adoc

```diff
@@ -0,0 +1,480 @@
+The type associated with a property must match the `type` attribute used in the corresponding `metatype.xml` file when applicable.
+
+The following table shows the format to use for each property type supported.
+
+[cols="1,2,4" options="header"]
+|===
+
+|Type
+|Format
+|Example
+
+|Service PID
+|`service.pid` = "servicePid"
+|`service.pid = "org.codice.${ddf-branding-lowercase}.security.policy.context.impl.PolicyManager"`
+
+|Factory PID
+|`service.factoryPid` = "serviceFactoryPid"
+|`service.factoryPid = "Csw_Federated_Source"`
+
+|Strings
+|`name = "value"`
+|`name = "john"`
+
+|Booleans
+|`name = B"true\|false"`
+|`authorized = B"true"`
+
+|Integers
+|`name = I"value"`
+|`timeout=I"60"`
+
+|Longs
+|`name = L"value"`
+|`diameter = L"10000"`
+
+|Floats
+|`name = F"value"`
+|`cost = F"10.50"`
+
+|Doubles
+|`name = D"value"`
+|`latitude = D"45.0234"`
+
+|Lists of Strings
+|`name = [ "value1", "value2", ... ]`
+|`authenticationTypes = [ "/\=SAML\|GUEST", "/admin\=SAML\|basic", "/jolokia\=SAML\|basic", "/system\=basic", "/solr\=SAML\|PKI\|basic", "/sources\=SAML\|basic", "/security-config\=SAML\|basic" ]`
+
+|Lists of Integers
+|`name = I[ "value1", "value1", ... ]`
+|`sizes = I[ "10", "20", "30" ]`
+
+|===
+
+[NOTE]
+====
+* Lists of values can be prefixed with any of the supported types (`B`, `I`, `L`, `F` or `D`)
+* To prevent any configuration issues, the `=` signs used in values should be escaped using `\`
+* Boolean values will default to `false` if any value other than `true` is provided
+====
+
+.Sample configuration file
+[source,java,linenums]
+----
+service.pid="org.codice.${ddf-branding-lowercase}.security.policy.context.impl.PolicyManager"
+
+authenticationTypes=["/\=SAML|GUEST","/admin\=SAML|basic","/jolokia\=SAML|basic","/system\=basic","/solr\=SAML|PKI|basic","/sources\=SAML|basic","/security-config\=SAML|basic","/search\=basic"]
+
+realms=["/\=karaf"]
+
+requiredAttributes=["/\=","/admin\={http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role\=admin}","/solr\={http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role\=admin}","/jolokia\={http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role\=admin}","/system\={http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role\=admin}","/security-config\={http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role\=admin}"]
+
+whiteListContexts=["/services/SecurityTokenService","/services/internal/metrics","/services/saml","/proxy","/services/csw"]
+----
+
+====== Editing HTTP Ports for Multiple Local ${branding} Nodes
+
+Edit the port numbers in the files in the ${branding} install folder.
+
+[cols="4" options="header"]
+|===
+
+|File to Edit
+|Property(ies)
+|Original Value
+|Example of New Value
+
+|`bin/karaf.bat`
+|`address`
+|`5005`
+|i.e. `5006`
+
+|`etc/org.apache.karaf.management.cfg`
+|`rmiRegistryPort`
+|`1099`
+|i.e. `1199`
+
+|" "
```

Comment:
I think I'll change it to merged cells. That will get the point across much better.